src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -12,19 +1,59 @@
+import { Dispatch, SetStateAction } from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+import { SetStateСompletedStatus } from '../../types/SetStateСompletedStatus';
+import { СompletedStatus } from '../../types/СompletedStatus';
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<{
+  query: string;
+  setQuery: Dispatch<SetStateAction<string>>;
+  setFilterStatus: SetStateСompletedStatus;
+}> = ({ query, setQuery, setFilterStatus }) => {
+  function handleSearchChange(value: string) {
+    setQuery(value);
+  }
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            onChange={event =>
+              setFilterStatus(event.target.value as СompletedStatus)
```

Comment:
Ensure that the value from `event.target.value` is correctly cast to the `СompletedStatus` type. This is important to prevent runtime errors if the value does not match the expected enum type.